File: javascript/0003_IQ_Test.js
function iqTest(numbers) {
  const nums = numbers.split(" ");

  let odds = [];
  let evens = [];

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] % 2 != 0) odds.push(i + 1);
    else evens.push(i + 1);
  }

  return odds.length > evens.length ? evens[0] : odds[0];
}

console.log(iqTest("2 4 7 8 10"));
console.log(iqTest("1 2 2"));

File: python/0003_IQ_Test.py
def isodd(number):
  return True if int(number)%2 != 0 else False

def iq_test(numbers):
  mynumbers = numbers.split(" ")
  i = 0
  countOdd = 0
  countEven = 0
  diff = 0
  for number in mynumbers:
    isit = isodd(number)

    if isit:
      countOdd+=1
    else:
      countEven+=1
  i=0
  for number in mynumbers:
    isit = isodd(number)
    if countOdd > countEven:
      if not isit:
        diff = i
    elif countEven > countOdd:
      if isit:
        diff = i
    #print(str(isit) + " : " + number)
    i+=1
  return(diff+1)